internal/http/url_handler_test.go

```diff
@@ -0,0 +1,85 @@
+package http
+
+import (
+	"context"
+	"github.com/Kenny201/go-yandex-shortener.git/internal/app/shortener"
+	"github.com/go-chi/chi/v5"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestPostHandler(t *testing.T) {
+	req, err := http.NewRequest(http.MethodPost, "http://localhost:8080/", strings.NewReader("https://practicum.yandex.ru"))
+
+	if err != nil {
+		t.Fatalf("method not alowed: %v", err)
+	}
+
+	w := httptest.NewRecorder()
+
+	ss := shortener.NewService(shortener.WithRepositoryMemory())
+
+	NewShortenerHandler(ss).PostHandler(w, req)
+
+	res := w.Result()
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusCreated {
+		t.Errorf("excpected statud OK; got %v", res.StatusCode)
+	}
+
+	if ctype := w.Header().Get("Content-Type"); ctype != "text/plain" {
+		t.Errorf("content type header does not match: got %v want %v",
+			ctype, "text/plain")
+	}
+
+	_, err = io.ReadAll(res.Body)
```

Comment:
Не стоит забывать закрывать тело ответа даже в тестах.

```suggestion
	_, err = io.ReadAll(res.Body)
	defer res.Body.Close()
```